test/lsp.py

```diff
@@ -1534,6 +1541,70 @@ def test_custom_includes(self, solc: JsonRpcProcess) -> None:
         self.expect_equal(len(diagnostics), 1, "no diagnostics")
         self.expect_diagnostic(diagnostics[0], code=2018, lineNo=5, startEndColumns=(4, 62))
 
+    def test_remapping_wrong_paramateres(self, solc: JsonRpcProcess) -> None:
+        self.setup_lsp(solc, expose_project_root=False)
+        solc.send_notification(
+            'workspace/didChangeConfiguration', {
+                'settings': {
+                    'remappings': [
+                        {
+                            'context': False,
+                            'prefix': 1,
+                            'target': f"{self.project_root_dir}/other-include-dir/otherlib/"
+                        }
+                    ]
+                }
+            }
+        )
+
+        response = solc.receive_message()
+        self.expect_equal(response["method"], "$/logTrace")
+
+        if not response["params"]["message"].startswith("Invalid JSON configuration passed."):
+            raise Exception("Expected JSON error.")
+
+    def test_remapping(self, solc: JsonRpcProcess) -> None:
+        self.setup_lsp(solc, expose_project_root=False)
+        solc.send_notification(
+            'workspace/didChangeConfiguration', {
+                'settings': {
+                    'remappings': [
+                        {
+                            'context': '',
+                            'prefix': '@other/',
+                            'target': f"{self.project_root_dir}/other-include-dir/otherlib/"
+                        }
+                    ]
+                }
+            }
+        )
+
+        # test file
+        file_with_remapped_import_uri = 'file:///remapped-include.sol'
```

Comment:
It would be best to set this to something that would match the remapping if used in an import to show that these URIs are not remapped.